lib/archethic/contracts/interpreter.ex

```diff
@@ -153,18 +137,39 @@ defmodule Archethic.Contracts.Interpreter do
 
       trigger_code ->
         timestamp_now =
-          time_now(trigger_type, maybe_trigger_tx)
+          case Keyword.get(opts, :time_now) do
+            nil ->
+              # you must use the :time_now opts during the validation workflow
+              # because there is no validation_stamp yet
+              time_now(trigger_type, maybe_trigger_tx)
```

Comment:
This could be as the default value of `Keyword.get` : `Keyword.get(opts, :time_now, time_now(trigger_type, maybe_trigger_tx))`

Reply:
nope it's not possible because the `time_now` cannot be executed if there is no `validation_stamp`. It'd raise.